src/integration_tests/schemas/e2e_schema_integration_test.py

```diff
@@ -192,3 +195,231 @@ def test_survey_id_map(self):
 
         assert set_survey_id_map_response.status_code == 200
         assert set_survey_id_map_response.json() == test_survey_id_map
+
+
+    @pytest.mark.order(6)
+    def test_post_schema_unauthorized(self):
+        """
+        Test unauthorized access by providing incorrect authorization token for POST /v1/schema.
+
+        * Send a request to POST /v1/schema with an invalid token.
+        * Assert status code: 401 Unauthorized.
+        """
+        response = self.session.post(
+            f"{config.API_URL}/v1/schema?survey_id={test_survey_id}",
+            json=invalid_data,
+            headers=self.invalid_token_headers,
+        )
+        assert response.status_code == 401
+        if is_json_response(self, response) and "detail" in response.json():
+            assert response.json()["detail"] == "Unauthorized access"
+        else:
+            print("Non-JSON Response:", response.text) 
+
+        
+    @pytest.mark.order(7)
+    def test_post_schema_validation_error(self):
+        """
+        Test validation issue by providing invalid data for POST /v1/schema.
+
+        * We test the POST /v1/schema endpoint by providing invalid data.
+        * Assert status code: 400 Bad Request.
+        """
+        response = self.session.post(
+            f"{config.API_URL}/v1/schema?survey_id={test_survey_id}",
+            json=invalid_data,
+            headers=self.headers,
+        )
+        assert response.status_code == 400
+        if "detail" in response.json():
+            assert response.json()["detail"] == "Invalid data format"
+        else:
+            print("Response JSON:", response.json())
```

Comment:
Conditional block should not exist in test assertion unless it is to act upon the configuration.
This block is not going to invalidate a response such as
`{"message": "Invalid data format"}` or `{"detail": "abc"}`
Because the `else` block allow the code to continue without any interruption
** Printing will have no effect when running test. If you need to log anything, use the logger

Change the conditional block into the following to properly invalidate a response:
`assert is_json_response(response)`
`assert "detail" in response.json()`
`assert response.json()["detail"] == "Invalid data format"`